dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/validation/validator/event/DataRelationsValidatorTest.java

```diff
@@ -750,6 +757,113 @@ void eventValidationFailsWhenEventAOCAndEventCOsAreSetAndInProgramCCButNotAllCOs
                     && r.getMessage().contains(aoc.getUid())));
   }
 
+  @Test
+  void eventOfProgramWithRegistrationInEnrollmentWithoutTrackedEntity() {
+    OrganisationUnit orgUnit = organisationUnit(ORG_UNIT_ID);
+    when(preheat.getOrganisationUnit(MetadataIdentifier.ofUid(ORG_UNIT_ID))).thenReturn(orgUnit);
+
+    Program program = setupProgram(orgUnit);
+
+    CategoryCombo defaultCC = defaultCategoryCombo();
+    program.setCategoryCombo(defaultCC);
+    CategoryOptionCombo defaultAOC = firstCategoryOptionCombo(defaultCC);
+    when(preheat.getDefault(CategoryOptionCombo.class)).thenReturn(defaultAOC);
+
+    Event event = eventBuilder().enrollment(ENROLLMENT_ID).build();
+
+    when(preheat.getEnrollment(ENROLLMENT_ID)).thenReturn(null);
```

Comment:
that is the default right? any setup line we can save is great 😅

Reply:
no, the default is returning the enrollment in `setupProgram` so we need to override with null